### PR TITLE
Fix code smells; Fix #395

### LINKED ISSUE #395: SingleIterationApplicationViewModel : Split record change method
### Prerequisites

- [x] I have written a descriptive issue title
- [x] I have verified that I am running the latest version of the COMET-WEB
- [x] I have searched [open](https://github.com/RHEAGROUP/COMET-WEB-Community-Edition/issues) and [closed](https://github.com/RHEAGROUP/COMET-WEB-Community-Edition/issues?q=is%3Aissue+is%3Aclosed) issues to ensure it has not already been reported

### Description

For now, the RecordChange method, inside the SingleIterationApplicationViewModel pre-filters to check if the container of the Iteration is the same. This filtering should be part of an overridable method and remove the virtual on the RecordChange method. 

With that, classes that implements that VM can customize the way of filtering or totally remove the filtering process.
### Logs

  - [ ] There are no logs regarding the issue
  - [ ] I have attached a log file or included the relevant section in this issue  
  
### System Configuration

- COMET version:
  - [ ] COMET-WEB: 
  - [ ] Other: 
- Web Services:
  - [ ] COMET Web Services version:

### PR DESCRIPTION
**Pull request by jaimeatrhea**
### Prerequisites

- [x] I have written a descriptive pull-request title
- [x] I have verified that there are no overlapping [pull-requests](https://github.com/RHEAGROUP/COMET-WEB-Community-Edition/pulls) open
- [x] I have verified that I am following the COMET-WEB [code style guidelines](https://raw.githubusercontent.com/RHEAGROUP/COMET-WEB-Community-Edition/master/.github/CONTRIBUTING.md)
- [x] I have provided test coverage for my change (where applicable)

### Description
This PR fixes some code smells in the javascript files. It fixes #395 and it uses that logic in the ParameterEditorBodyViewModel

### CODE REVIEW

File: COMETwebapp/wwwroot/Scripts/babylonSpecifics.js [hunk 1]
```diff
@@ -42,7 +42,7 @@ function CreateScene(engine, canvas) {
     Camera.panningSensibility = CameraPanningSensibility;
     Camera.wheelPrecision = CameraZoomSensibility;
 
-    let light1 = new BABYLON.HemisphericLight("HemisphericLight", new BABYLON.Vector3(2, 1, 0), scene);
```
Comment (antoineatrhea) [inline COMETwebapp/wwwroot/Scripts/babylonSpecifics.js:45]: This produce a bug detected by SonarQube, can you fix it please?

**Review by antoineatrhea — CHANGES_REQUESTED** [review]
Fix SonarQube please

**Review by nabilatrhea — APPROVED** [review]
after fixing the detected Bug

**Review by samatrhea — APPROVED** [review]

**Comment by samatrhea** [comment]
@jaimeatrhea any update on this?

**Comment by samatrhea** [comment]
@jaimeatrhea any more work to be done on this?

**Comment by jaimeatrhea** [comment]
@samatrhea No, unless the bug can de disabled from SonarQube

**Comment by antoineatrhea** [comment]
@samatrhea I think that we should exclude _.js_ files from SonarQube checks. I'm afraid that resolving #420 will bring too many SonarQube issues.

**Review by omaratrhea — APPROVED** [review]

**Review by samatrhea — APPROVED** [review]

**Pull request merged**